models/object_detection_yolox/demo.cpp

```diff
@@ -0,0 +1,325 @@
+
+        int x0 = box.at<int>(0, 0);
+        int y0 = box.at<int>(0, 1);
+        int x1 = box.at<int>(0, 2);
+        int y1 = box.at<int>(0, 3);
+
+        string text = format("%s : %f", labelYolox[clsId].c_str(), score * 100);
+        int font = FONT_HERSHEY_SIMPLEX;
+        int baseLine = 0;
+        Size txtSize = getTextSize(text, font, 0.4, 1, &baseLine);
+        rectangle(resImg, Point(x0, y0), Point(x1, y1), Scalar(0, 255, 0), 2);
+        rectangle(resImg, Point(x0, y0 + 1), Point(x0 + txtSize.width + 1, y0 + int(1.5 * txtSize.height)), Scalar(255, 255, 255), -1);
+        putText(resImg, text, Point(x0, y0 + txtSize.height), font, 0.4, Scalar(0, 0, 0), 1);
+    }
+
+    return resImg;
+}
+
+int main(int argc, char** argv)
+{
+    CommandLineParser parser(argc, argv, keys);
+
+    parser.about("Use this script to run Yolox deep learning networks in opencv_zoo using OpenCV.");
+    if (parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    string model = parser.get<String>("model");
+    float confThreshold = parser.get<float>("confidence");
+    float objThreshold = parser.get<float>("obj");
+    float nmsThreshold = parser.get<float>("nms");
+    bool vis = parser.get<bool>("vis");
+    bool save = parser.get<bool>("save");
+    int backendTargetid = parser.get<int>("backend");
+
+    if (model.empty())
+    {
+        CV_Error(Error::StsError, "Model file " + model + " not found");
+    }
+
+    YoloX modelNet(model, confThreshold, nmsThreshold, objThreshold,
+        backendTargetPairs[backendTargetid].first, backendTargetPairs[backendTargetid].second);
+    //! [Open a video file or an image file or a camera stream]
+    VideoCapture cap;
+    if (parser.has("input"))
+        cap.open(samples::findFile(parser.get<String>("input")));
+    else
+        cap.open(0);
+    if (!cap.isOpened())
+        CV_Error(Error::StsError, "Cannot opend video or file");
+    Mat frame, inputBlob;
+    double letterboxScale;
+
+    static const std::string kWinName = model;
+    int nbInference = 0;
+    while (waitKey(1) < 0)
+    {
+        cap >> frame;
+        if (frame.empty())
+        {
+            cout << "Frame is empty" << endl;
+            waitKey();
+            break;
+        }
+        pair<Mat, double> w = letterBox(frame);
+        inputBlob = get<0>(w);
+        letterboxScale  = get<1>(w);
+        TickMeter tm;
+        tm.start();
+        Mat predictions = modelNet.infer(inputBlob);
+        tm.stop();
+        cout << "Inference time: " << tm.getTimeMilli() << " ms\n";
+        Mat img = visualize(predictions, frame, letterboxScale, tm.getFPS());
+        if (vis)
+        {
+            imshow(kWinName, img);
+        }
+        else
+        {
+            nbInference++;
+            if (nbInference > 100)
+            {
+                cout << nbInference << " inference made. Demo existing" << endl;
+                break;
+            }
```

Comment:
I dont think we should make any assumption that the user want to run certain times of inference and then exit. Pressing a key to exit is enough.